spotify_player/src/cli/client.rs

```diff
@@ -350,3 +362,359 @@ async fn handle_playback_request(
     client.update_playback(state);
     Ok(())
 }
+
+async fn handle_playlist_request(
+    client: &Client,
+    state: &SharedState,
+    command: PlaylistCommand,
+) -> Result<String> {
+    match command {
+        PlaylistCommand::New {
+            name,
+            public,
+            collab,
+            description,
+        } => {
+            let user = state.data.read().user_data.user.to_owned().unwrap();
+            let id = user.id;
+
+            let resp = client
+                .spotify
+                .user_playlist_create(
+                    id,
+                    name.as_str(),
+                    Some(public),
+                    Some(collab),
+                    Some(description.as_str()),
+                )
+                .await?;
+            Ok(format!(
+                "Playlist '{}' with id '{}' was created.",
+                resp.name, resp.id
+            ))
+        }
+        PlaylistCommand::Delete { id } => {
+            let user = state.data.read().user_data.user.to_owned().unwrap();
+            let uid = user.id;
+
+            let playlist_check = client
+                .spotify
+                .playlist_check_follow(id.to_owned(), &[uid])
+                .await;
+            if playlist_check.is_err() {
+                anyhow::bail!("Could not find playlist {}", id)
+            }
+
+            // Won't delete if not following
+            let following = playlist_check.unwrap().pop().unwrap();
+            if following {
+                client.spotify.playlist_unfollow(id.to_owned()).await?;
+                Ok(format!("Playlist '{}' was deleted/unfollowed", id))
+            } else {
+                Ok(format!(
+                    "Playlist '{}' was not followed by the user, nothing to be done.",
+                    id
+                ))
+            }
+        }
+        PlaylistCommand::List => {
+            let resp = client.current_user_playlists().await?;
+
+            let mut out = String::new();
+            for pl in resp {
+                // Might want to add color
+                out.push_str(format!("{}: {}\n", pl.id.id(), pl.name).as_str());
+            }
+            out = out.trim().to_string();
+
+            Ok(out)
+        }
+        PlaylistCommand::Import {
+            from: import_from,
+            to: import_to,
+            delete,
+        } => playlist_import(client, import_from, import_to, delete).await,
+        PlaylistCommand::Fork { id } => {
+            let user = state.data.read().user_data.user.to_owned().unwrap();
+            let uid = user.id;
+
+            let from = client
+                .spotify
+                .playlist(id.to_owned(), None, None)
+                .await
+                .unwrap_or_else(|_| panic!("Cannot import from {}. Playlist not found.", id.id()));
```

Comment:
```suggestion
            let from = client
                .spotify
                .playlist(id.to_owned(), None, None)
                .await?;
```

Reply:
This can fail for many other reasons than playlist not found. The propagated error's message should be enough to identify it, so no need to add additional context.